mobile/lib/modules/favorite/providers/favorite_provider.dart

```diff
@@ -0,0 +1,50 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/shared/providers/asset.provider.dart';
+import 'package:openapi/api.dart';
+
+class FavoriteSelectionNotifier extends StateNotifier<Set<String>> {
```

Comment:
This was necessary for performance reasons. Updating the assetProvider all the time was too slow.
This solution also works in albums.

Reply:
Looks good!